auction-server/src/opportunity/repository/add_opportunity.rs

```diff
@@ -30,28 +30,35 @@ impl<T: InMemoryStore> Repository<T> {
             .expect("creation_time is valid");
         let metadata = opportunity.get_models_metadata();
         let chain_type = <T::Opportunity as entities::Opportunity>::ModelMetadata::get_chain_type();
-        sqlx::query!("INSERT INTO opportunity (id,
-                                                        creation_time,
-                                                        permission_key,
-                                                        chain_id,
-                                                        chain_type,
-                                                        metadata,
-                                                        sell_tokens,
-                                                        buy_tokens) VALUES ($1, $2, $3, $4, $5, $6, $7, $8)",
-        opportunity.id,
-        PrimitiveDateTime::new(odt.date(), odt.time()),
-        opportunity.permission_key.to_vec(),
-        opportunity.chain_id,
-        chain_type as _,
-        serde_json::to_value(metadata).expect("Failed to serialize metadata"),
-        serde_json::to_value(&opportunity.sell_tokens).unwrap(),
-        serde_json::to_value(&opportunity.buy_tokens).unwrap())
-            .execute(db)
-            .await
-            .map_err(|e| {
-                tracing::error!("DB: Failed to insert opportunity: {}", e);
-                RestError::TemporarilyUnavailable
-            })?;
+        if self
```

Comment:
I think we can have sth like this:

`get_or_add_db_opportunity`
`exists_db_opportunity`
`add_db_opportunity`

While the opportunity dont have any unique key, I think we dont need transaction for the query. So I think the above structure will be sufficient for our case.